src/java/org/apache/cassandra/index/sai/utils/RangeIntersectionIterator.java

```diff
@@ -93,9 +93,6 @@ protected PrimaryKey computeNext()
                         // the other iterators except this one to match the new highest key.
                         continue outer;
                     }
-                    assert nextKey.compareTo(highestKey) == 0:
```

Comment:
Are we removing this assertion because it is no longer true?

Reply:
because compareTo is super expensive

Reply:
Makes sense. Why don't we switch it to:

```java
                    int comparisonResult;
                    if (nextKey == null || (comparisonResult = nextKey.compareTo(highestKey)) > 0)
                    {
                        // We jumped over the highest key seen so far, so make it the new highest key.
                        highestKey = nextKey;
                        // Remember this iterator to avoid advancing it again, because it is already at the highest key
                        alreadyAvanced = index;
                        // This iterator jumped over, so the other iterators are lagging behind now,
                        // including the ones already advanced in the earlier cycles of the inner loop.
                        // Therefore, restart the inner loop in order to advance
                        // the other iterators except this one to match the new highest key.
                        continue outer;
                    }
                    assert comparisonResult == 0:
                    String.format("skipTo skipped to an item smaller than the target; " +
                                  "iterator: %s, target key: %s, returned key: %s", range, highestKey, nextKey);
```

The bug we're tracking down right now is exposed by this assertion.